ctools/generate_fragmented_sharded_collection.py

```diff
@@ -26,7 +26,8 @@ async def main(args):
     cluster = Cluster(args.uri, asyncio.get_event_loop())
     await cluster.check_is_mongos(warn_only=False)
 
-    shard_key_as_string = True if (await cluster.FCV == '4.0') else False
+    fcv = await cluster.FCV
+    shard_key_as_string = True if (fcv == '4.0' or fcv == '4.2') else False
```

Comment:
I remember I did some testing and it should also work the following simpler version
```suggestion
    shard_key_as_string = fcv <= '4.2'
```
But I suggest to double test it if you decide to change it.